tests/UKHO.S100PermitService.Common.UnitTests/Services/PermitServiceTests.cs

```diff
@@ -261,106 +280,185 @@ public async Task WhenHoldingsServiceReturnsNotFoundResponse_ThenPermitServiceRe
            ).MustNotHaveHappened();
         }
 
-        private static (HttpStatusCode, List<HoldingsServiceResponse>?) GetHoldingDetails(string responseType)
+        [TestCase(BadRequest)]
+        public async Task WhenHoldingsServiceReturnsBadRequestResponse_ThenPermitServiceReturnsBadRequestResponse(string responseType)
+        {
+            A.CallTo(() => _fakeUserPermitService.GetUserPermitAsync(A<int>.Ignored, A<CancellationToken>.Ignored, A<string>.Ignored))
+                .Returns(GetUserPermits(OkResponse));
+
+            A.CallTo(() => _fakeHoldingsService.GetHoldingsAsync(A<int>.Ignored, A<CancellationToken>.Ignored, A<string>.Ignored))
+                .Returns(GetHoldingDetails(responseType));
+
+            var (httpResponseMessage, stream) = await _permitService.ProcessPermitRequestAsync(1, CancellationToken.None, _fakeCorrelationId);
+
+            httpResponseMessage.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+            httpResponseMessage.Content.ReadAsStringAsync().Result.Should().Be(ErrorHoldingsBadRequestContent);
+            stream.Length.Should().Be(0);
+
+            A.CallTo(() => _fakeHoldingsService.FilterHoldingsByLatestExpiry(A<List<HoldingsServiceResponse>>.Ignored)).MustNotHaveHappened();
+
+            A.CallTo(_fakeLogger).Where(call =>
+                call.Method.Name == "Log"
+                && call.GetArgument<LogLevel>(0) == LogLevel.Information &&
+                call.GetArgument<EventId>(1) == EventIds.ProcessPermitRequestStarted.ToEventId()
+                && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Process permit request started."
+            ).MustHaveHappenedOnceExactly();
+
+            A.CallTo(_fakeLogger).Where(call =>
+                call.Method.Name == "Log"
+                && call.GetArgument<LogLevel>(0) == LogLevel.Information
+                && call.GetArgument<EventId>(1) == EventIds.ProcessPermitRequestCompleted.ToEventId()
+                && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!
+                    .ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Process permit request completed."
+            ).MustNotHaveHappened();
+        }
+
+        private static (HttpResponseMessage, List<HoldingsServiceResponse>?) GetHoldingDetails(string responseType)
         {
             switch(responseType)
             {
                 case OkResponse:
-                    return (HttpStatusCode.OK,
+                    return (new HttpResponseMessage()
+                    {
+                        StatusCode = HttpStatusCode.OK
+                    },
                     [
                         new HoldingsServiceResponse
                         {
-                            ProductTitle = "ProductTitle",
-                            ProductCode = "ProductCode",
+                            UnitTitle = "ProductTitle",
```

Comment:
HoldingServiceResponse fields should be in Camel Case e.g. unitTitle, unitName please check all responses in this class.

Reply:
Resolved